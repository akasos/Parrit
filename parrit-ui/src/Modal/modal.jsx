import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'

const Modal = (props) => {
    return ReactDOM.createPortal(
        props.children,
        props.domElement
    );
};

Modal.propTypes = {
    children: PropTypes.node.isRequired,
    domElement: PropTypes.object
};
export default Modal;