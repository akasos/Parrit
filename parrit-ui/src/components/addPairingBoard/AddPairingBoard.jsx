import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import style from "styled-components";
import {addPairingBoard} from "../actions";

const ModalContainer = style.div``;
const AddPairingBoardContainer = style.div``;
const TitleWrapper = style.div``;
const Title = style.h2``;
const AddPairingBoardInput = style.input``;
const Button = style.button``;


export class AddPairingBoard extends Component {
    constructor(props) {
        super(props);
        this.addPairingBoard = this.addPairingBoard.bind(this);
    }

    state = {
        title: ''
    };

    onInputChange = event => {
        this.setState({title: event.target.value});
    };

    addPairingBoard() {
    this.props.addPairingBoard(this.state);
    this.props.cancel();
    };

    render() {
        const {cancel} = this.props;
        return (
            <ModalContainer data-testid="test" className="ui dimmer modals visible active modal-container"
                            onClick={cancel}>
                <AddPairingBoardContainer onClick={(event) => event.stopPropagation()}
                                          className="ui standard modal visible active">
                    <TitleWrapper><Title>Add Pairing Board</Title></TitleWrapper>
                    <AddPairingBoardInput className="add-pairing-board-input" type="text" onChange={this.onInputChange}
                                          value={this.state.name}/>
                    <div className="actions">
                        <Button onClick={this.addPairingBoard}
                                className="ui primary button add-pairing-board-button">Ok</Button>
                        <Button onClick={cancel} className="ui button cancel-adding-teammate-button">Cancel</Button>
                    </div>
                </AddPairingBoardContainer>
            </ModalContainer>

        );
    }
}

AddPairingBoard.propTypes = {
    cancel: PropTypes.func.isRequired,
    addPairingBoard: PropTypes.func.isRequired
};

export default connect(null, {addPairingBoard})(AddPairingBoard);