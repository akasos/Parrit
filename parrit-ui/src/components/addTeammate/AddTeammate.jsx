import React, {Component} from 'react';
import style from 'styled-components'
const ModalContainer = style.div``;
const AddTeamMateContainer = style.div``;
const TitleWrapper = style.div``;
const Title = style.h2``;
const AddTeammateInput = style.input``;
const Button = style.button``;

class AddTeammate extends Component {
    state = {
      name: ''
    };

    onInputChange  = (event) => {
        this.setState({name: event.target.value});
    };

    render() {
        const { cancel } = this.props;
        return (
            <ModalContainer className="ui dimmer modals visible active modal-container" onClick={cancel}>
                <AddTeamMateContainer onClick={(event) => event.stopPropagation()} className="ui standard modal visible active">
                    <TitleWrapper><Title>Add Parrit Teammate</Title></TitleWrapper>
                    <AddTeammateInput className="add-teammate-input" type="text" onChange={this.onInputChange} value={this.state.name}/>
                    <div className ="actions">
                        <Button className="ui primary button">Ok</Button>
                        <Button className="ui button">Cancel</Button>
                    </div>
                </AddTeamMateContainer>
            </ModalContainer>
        );
    }
}

export default AddTeammate;