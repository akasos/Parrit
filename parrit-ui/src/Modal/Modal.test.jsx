import React from 'react';
import Modal from "./modal";
import {mount} from 'enzyme';
import {AddPairingBoard} from "../components/addPairingBoard/AddPairingBoard";
import {AddTeammate} from "../components/addTeammate/AddTeammate";


function renderModal(args) {
    const props = {domElement: document.createElement('div')};
    return mount(<Modal {...props}>{args}</Modal>);
}

describe("Modal", () => {
    it('should render the `AddTeammate` component', () => {
        const component1 = renderModal(<AddTeammate cancel={jest.fn()} addTeammmate={jest.fn()}/>);
        expect(component1.debug()).toMatchSnapshot();
    });
    it('should render the `AddPairingBoard` component', () => {
        const component2 = renderModal(<AddPairingBoard cancel={jest.fn()} addTeammmate={jest.fn()}/>);
        expect(component2.debug()).toMatchSnapshot();

    });

});


