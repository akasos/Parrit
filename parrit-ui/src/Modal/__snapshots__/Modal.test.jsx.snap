// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render the \`AddPairingBoard\` component 1`] = `
"<Modal domElement={{...}}>
  <Portal containerInfo={{...}}>
    <AddPairingBoard cancel={[Function: mockConstructor]} addPairingBoard={[Function: mockConstructor]}>
      <styled.div data-testid=\\"test\\" className=\\"ui dimmer modals visible active modal-container\\" onClick={[Function: mockConstructor]}>
        <StyledComponent data-testid=\\"test\\" className=\\"ui dimmer modals visible active modal-container\\" onClick={[Function: mockConstructor]} forwardedComponent={{...}} forwardedRef={{...}}>
          <div data-testid=\\"test\\" className=\\"sc-bdVaJa bDWFJH ui dimmer modals visible active modal-container\\" onClick={[Function: mockConstructor]}>
            <styled.div onClick={[Function: onClick]} className=\\"ui standard modal visible active\\">
              <StyledComponent onClick={[Function: onClick]} className=\\"ui standard modal visible active\\" forwardedComponent={{...}} forwardedRef={{...}}>
                <div onClick={[Function: onClick]} className=\\"sc-bwzfXH uTxCW ui standard modal visible active\\">
                  <styled.div>
                    <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                      <div className=\\"sc-htpNat KtFsv\\">
                        <styled.h2>
                          <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                            <h2 className=\\"sc-bxivhb bUZGRr\\">
                              Add Pairing Board
                            </h2>
                          </StyledComponent>
                        </styled.h2>
                      </div>
                    </StyledComponent>
                  </styled.div>
                  <styled.input className=\\"add-pairing-board-input\\" type=\\"text\\" onChange={[Function]} value={[undefined]}>
                    <StyledComponent className=\\"add-pairing-board-input\\" type=\\"text\\" onChange={[Function]} value={[undefined]} forwardedComponent={{...}} forwardedRef={{...}}>
                      <input className=\\"sc-ifAKCX cFlEyZ add-pairing-board-input\\" type=\\"text\\" onChange={[Function]} value={[undefined]} />
                    </StyledComponent>
                  </styled.input>
                  <div className=\\"actions\\">
                    <styled.button onClick={[Function: bound addPairingBoard]} className=\\"ui primary button add-pairing-board-button\\">
                      <StyledComponent onClick={[Function: bound addPairingBoard]} className=\\"ui primary button add-pairing-board-button\\" forwardedComponent={{...}} forwardedRef={{...}}>
                        <button onClick={[Function: bound addPairingBoard]} className=\\"sc-EHOje jvzENE ui primary button add-pairing-board-button\\">
                          Ok
                        </button>
                      </StyledComponent>
                    </styled.button>
                    <styled.button onClick={[Function: mockConstructor]} className=\\"ui button cancel-adding-teammate-button\\">
                      <StyledComponent onClick={[Function: mockConstructor]} className=\\"ui button cancel-adding-teammate-button\\" forwardedComponent={{...}} forwardedRef={{...}}>
                        <button onClick={[Function: mockConstructor]} className=\\"sc-EHOje jvzENE ui button cancel-adding-teammate-button\\">
                          Cancel
                        </button>
                      </StyledComponent>
                    </styled.button>
                  </div>
                </div>
              </StyledComponent>
            </styled.div>
          </div>
        </StyledComponent>
      </styled.div>
    </AddPairingBoard>
  </Portal>
</Modal>"
`;

exports[`Modal should render the \`AddTeammate\` component 1`] = `
"<Modal domElement={{...}}>
  <Portal containerInfo={{...}}>
    <AddTeammate cancel={[Function: mockConstructor]} addTeammate={[Function: mockConstructor]}>
      <styled.div className=\\"ui dimmer modals visible active modal-container\\" onClick={[Function: mockConstructor]}>
        <StyledComponent className=\\"ui dimmer modals visible active modal-container\\" onClick={[Function: mockConstructor]} forwardedComponent={{...}} forwardedRef={{...}}>
          <div className=\\"sc-bZQynM bOrPFB ui dimmer modals visible active modal-container\\" onClick={[Function: mockConstructor]}>
            <styled.div onClick={[Function: onClick]} className=\\"ui standard modal visible active\\">
              <StyledComponent onClick={[Function: onClick]} className=\\"ui standard modal visible active\\" forwardedComponent={{...}} forwardedRef={{...}}>
                <div onClick={[Function: onClick]} className=\\"sc-gzVnrw ibZhkv ui standard modal visible active\\">
                  <styled.div>
                    <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                      <div className=\\"sc-htoDjs bZzhDu\\">
                        <styled.h2>
                          <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                            <h2 className=\\"sc-dnqmqq bDsesi\\">
                              Add Parrit Teammate
                            </h2>
                          </StyledComponent>
                        </styled.h2>
                      </div>
                    </StyledComponent>
                  </styled.div>
                  <styled.input className=\\"add-teammate-input\\" type=\\"text\\" onChange={[Function]} value=\\"\\">
                    <StyledComponent className=\\"add-teammate-input\\" type=\\"text\\" onChange={[Function]} value=\\"\\" forwardedComponent={{...}} forwardedRef={{...}}>
                      <input className=\\"sc-iwsKbI jQhHGA add-teammate-input\\" type=\\"text\\" onChange={[Function]} value=\\"\\" />
                    </StyledComponent>
                  </styled.input>
                  <div className=\\"actions\\">
                    <styled.button onClick={[Function: onClick]} className=\\"ui primary button add-teammate-button\\">
                      <StyledComponent onClick={[Function: onClick]} className=\\"ui primary button add-teammate-button\\" forwardedComponent={{...}} forwardedRef={{...}}>
                        <button onClick={[Function: onClick]} className=\\"sc-gZMcBi giNnHm ui primary button add-teammate-button\\">
                          Ok
                        </button>
                      </StyledComponent>
                    </styled.button>
                    <styled.button onClick={[Function: mockConstructor]} className=\\"ui button cancel-adding-teammate-button\\">
                      <StyledComponent onClick={[Function: mockConstructor]} className=\\"ui button cancel-adding-teammate-button\\" forwardedComponent={{...}} forwardedRef={{...}}>
                        <button onClick={[Function: mockConstructor]} className=\\"sc-gZMcBi giNnHm ui button cancel-adding-teammate-button\\">
                          Cancel
                        </button>
                      </StyledComponent>
                    </styled.button>
                  </div>
                </div>
              </StyledComponent>
            </styled.div>
          </div>
        </StyledComponent>
      </styled.div>
    </AddTeammate>
  </Portal>
</Modal>"
`;
